lib/https.js

```diff
@@ -152,41 +152,30 @@ Agent.prototype.createConnection = createConnection;
 Agent.prototype.getName = function getName(options) {
   var name = HttpAgent.prototype.getName.call(this, options);
 
-  name += ':';
-  if (options.ca)
-    name += options.ca;
-
-  name += ':';
-  if (options.cert)
-    name += options.cert;
-
-  name += ':';
-  if (options.clientCertEngine)
-    name += options.clientCertEngine;
-
-  name += ':';
-  if (options.ciphers)
-    name += options.ciphers;
-
-  name += ':';
-  if (options.key)
-    name += options.key;
-
-  name += ':';
-  if (options.pfx)
-    name += options.pfx;
-
-  name += ':';
-  if (options.rejectUnauthorized !== undefined)
-    name += options.rejectUnauthorized;
-
-  name += ':';
-  if (options.servername && options.servername !== options.host)
-    name += options.servername;
-
-  name += ':';
-  if (options.secureProtocol)
-    name += options.secureProtocol;
+  const components = [
+    options.ca,
+    options.cert,
+    options.clientCertEngine,
+    options.ciphers,
+    options.key,
+    options.pfx,
+    options.rejectUnauthorized,
+    options.servername !== options.host ? options.servername : undefined,
+    options.secureProtocol,
+    options.crl,
+    options.honorCipherOrder,
+    options.ecdhCurve,
+    options.dhparam,
+    options.secureOptions,
+    options.sessionIdContext
+  ];
+
+  for (const component of components) {
+    name += ':';
+    if (component != null) {
```

Comment:
It is better to be explicit and write `component !== null && component !== undefined` as that is faster than the implicit way. Using a plain for loop should also be a tad faster than the `for of` loop.

Reply:
`x != null` should in fact be faster than two strict checks but yes, for/of is probably a bit slower than a plain for loop.

What's more, creating a throwaway array like that has a lot of overhead compared to the code it replaces.  I expect you'll see a dip in the benchmarks.

Reply:
`x != null` was faster with Cranshaft. That is not true anymore since Turbofan.

And yes, the array is indeed quite a overhead. I would also rather prevent that but it makes the code here much more readable.

Reply:
This method is probably enough of a hot path that performance trumps readability.  Unless the benchmarks show otherwise, of course - to measure is to know (which doesn't have quite the same ring to it as "meten is weten" in Dutch, but I digress.)

> x != null was faster with Crankshaft. That is not true anymore since Turbofan.

It's a wash with TF (or at least it should be, the CSA code paths are practically identical) but `x != null` is faster with Ignition because it's just one dispatch from the interpreter loop instead of two (or three, if you include the `&&`.)

Probably still a wash in the grand scheme of things.  If nothing else, `x != null` is more succinct than the long-hand `x !== null && x !== undefined`.

Reply:
It sounds like the consensus is that collecting values into a temporary array isn't worth the performance hit, in which case the loop goes away anyway. If I remember correctly, the form with separate `if`s doesn't have any `x != null` checks either (it was added to the loop form to better account for numbers/booleans than a straight truthy/falsy check), so that point becomes moot as well. So, shall I flip it back to separate `if` checks?

Reply:
Don't have anything to add of value but re: the performance discussion, there's one notable case where `!= null` is significantly slower and that's when dealing with Objects (due to the possibility of `document.all`).